flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperator.java

```diff
@@ -442,36 +456,50 @@ public RetryableResultHandlerDelegator(
             this.processingTimeService = processingTimeService;
         }
 
-        public void registerTimeout(long timeout) {
-            resultHandler.registerTimeout(processingTimeService, timeout);
+        private void registerTimeout(long timeout) {
+            resultHandler.timeoutTimer =
+                    registerTimer(processingTimeService, timeout, t -> timerTriggered());
+        }
+
+        private void cancelRetryTimer() {
+            if (delayedRetryTimer != null) {
+                delayedRetryTimer.cancel(true);
```

Comment:
We may use false here to avoid interrupt the timer thread unexpectically and cause possible failures.